packages/backend/src/storage.ts

```diff
@@ -216,10 +214,30 @@ export default () => {
         },
     );
 
+    router.get('/new/:ensName/migrationStatus', async (req, res, next) => {
+        try {
+            const ensName = normalizeEnsName(req.params.ensName);
+            const status = await db.getUserDbMigrationStatus(ensName);
+            return res.json(status);
+        } catch (e) {
+            next(e);
+        }
+    });
+
+    router.post('/new/:ensName/migrationStatus', async (req, res, next) => {
+        try {
+            const ensName = normalizeEnsName(req.params.ensName);
+            await db.setUserDbMigrated(ensName);
```

Comment:
Where does the actual migration happen?

Reply:
Since the user data is encrypted the client has to to the migration. The flow would look like this

1. If old storage exists and not has been migrated
2. Get old Storage
3. Decrypt old Storage
4. Run Migration
5. Set the Migrated flag to true so the next time the app is opened the migration will not be triggered again.

Reply:
Ok, that makes perfect sense.